core/src/com/rallyhealth/weepickle/v1/core/Types.scala

```diff
@@ -114,8 +114,8 @@ trait Types { types =>
   trait From[In] {
     def narrow[K] = this.asInstanceOf[From[K]]
     def transform[Out](in: In, out: Visitor[_, Out]): Out = {
-      if (in == null) out.visitNull()
-      else transform0(in, out)
+      if (in == null) throw new NullPointerException()
```

Comment:
I'm not sure I like this part. Why not just call `visitNull` and let the visitor decide if it can handle it? (Which, in the case of most visitors extending `SimpleVisitor`, would throw a more descriptive `Abort` indicating what was expected and not an NPE.) Wouldn't it be less surprising for `FromScala(null).transform(to[Option[String]])` to return `None` instead of NPE?